internal/services/cognitive/cognitive_account_rai_policy_resource.go

```diff
@@ -0,0 +1,396 @@
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r CognitiveAccountRaiPolicyResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Cognitive.RaiPoliciesClient
+
+			id, err := raipolicies.ParseRaiPolicyID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			cognitiveAccountId := raipolicies.NewAccountID(id.SubscriptionId, id.ResourceGroupName, id.AccountName)
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			state := AccountRaiPolicyResourceModel{}
+
+			if model := resp.Model; model != nil {
+				state.Name = pointer.From(model.Name)
+				state.AccountId = cognitiveAccountId.ID()
+				state.Tags = pointer.From(model.Tags)
+
+				if props := model.Properties; props != nil {
+					state.BasePolicyName = pointer.From(props.BasePolicyName)
+					state.ContentFilter = flattenRaiPolicyContentFilters(props.ContentFilters)
+					state.CustomBlockList = flattenRaiCustomBlockLists(props.CustomBlocklists)
+					state.Mode = string(pointer.From(props.Mode))
+					state.Type = string(pointer.From(props.Type))
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r CognitiveAccountRaiPolicyResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Cognitive.RaiPoliciesClient
+
+			id, err := raipolicies.ParseRaiPolicyID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model AccountRaiPolicyResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			cognitiveAccountId := raipolicies.NewAccountID(id.SubscriptionId, id.ResourceGroupName, id.AccountName)
+
+			locks.ByID(cognitiveAccountId.ID())
+			defer locks.UnlockByID(cognitiveAccountId.ID())
+
+			props := resp.Model
+
+			if metadata.ResourceData.HasChange("content_filter") {
+				props.Properties.ContentFilters = expandRaiPolicyContentFilters(model.ContentFilter)
+			}
+
+			if metadata.ResourceData.HasChange("custom_blocklist") {
+				props.Properties.CustomBlocklists = expandCustomBlockLists(model.CustomBlockList)
+			}
+
+			if metadata.ResourceData.HasChange("mode") {
+				props.Properties.Mode = pointer.To(raipolicies.RaiPolicyMode(model.Mode))
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				props.Tags = pointer.To(model.Tags)
+			}
```

Comment:
This is not consistent with what we have outlined in the contributor guidelines or with the update methods of other resources in the provider. Please nil check model and properties properly and re-review t[his section of the guidelines](https://github.com/hashicorp/terraform-provider-azurerm/blob/66c60863f273a31ca6f36ed2ecf57b71fccb78b8/contributing/topics/guide-new-resource.md?plain=1#L259-L266) as well as [this example](https://github.com/hashicorp/terraform-provider-azurerm/blob/66c60863f273a31ca6f36ed2ecf57b71fccb78b8/internal/services/devcenter/dev_center_project_environment_type.go#L249-L260)

Reply:
Okay, will update this section